2022/form.py

```diff
@@ -85,6 +85,11 @@ def rowsum(self, rows):
             if r in self:
                 isNone = False
                 val += self[r]
+
+        for p in patterns:
+            pat = re.compile(p)
+            val += sum(v for k, v in self.data.items() if pat.match(k))
```

Comment:
We need to make sure to set isNone=False if there are any matches.

Reply:
Also should use `fullmatch()` because I think substring matches are not intended here.